Utilities/bootstrap

```diff
@@ -548,10 +555,15 @@ def get_swiftpm_flags(args):
             "-Xlinker", "@executable_path/../../../../../SharedFrameworks",
         ])
 
-    # Add a rpath to find core swift libraries on linux. We don't need a rpath
-    # for Darwin because the Swift libraries are present in the OS.
+    # Add a rpath to find core swift libraries on linux and Android. We don't
+    # need a rpath for Darwin because the Swift libraries are present in the OS.
     if platform.system() == 'Linux':
-        build_flags.extend(["-Xlinker", "-rpath=$ORIGIN/../lib/swift/linux"])
```

Comment:
We can probably drive this based on `swift -print-target-info` as well 🤔

Reply:
cc @DougGregor

Reply:
Yeah, ‘-print-target-info’ indicates whether one should use rpaths

Reply:
@aciidb0mb3r, do you mean extracting the relative path from the provided `runtimeLibraryPaths` instead of hardcoding it for each platform here? Because `librariesRequireRPath` is currently useless on [non-Darwin platforms, always returning false](https://github.com/apple/swift/blob/23b1a6b87afee535296a786dcdd2a6d0e65d67ef/lib/Basic/Platform.cpp#L57).

Reply:
Exacting the "linux" or "android" bit from the triple should be good enough, I think? @DougGregor or @brentdax might have better ideas though.

Anyway, I don't want to block your PR on this stuff so I am happy to take this patch and perhaps you can try `swift -print-target-info` in a follow on patch 😃

Reply:
Sounds good, I'll try it in my next pull.